src/Admin/AMPPlugins.php

```diff
@@ -0,0 +1,271 @@
+use AmpProject\AmpWP\Infrastructure\Delayed;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use stdClass;
+use function get_current_screen;
+
+/**
+ * Add new tab (AMP) in plugin install screen in WordPress admin.
+ *
+ * @since 2.2
+ * @internal
+ */
+class AMPPlugins implements Conditional, Delayed, Service, Registerable {
+
+	/**
+	 * Assets handle.
+	 *
+	 * @var string
+	 */
+	const ASSET_HANDLE = 'amp-plugin-install';
+
+	/**
+	 * @var array List of AMP plugins.
+	 */
+	public static $plugins = [];
+
+	/**
+	 * Get the action to use for registering the service.
+	 *
+	 * @return string Registration action to use.
+	 */
+	public static function get_registration_action() {
+
+		return 'current_screen';
+	}
+
+	/**
+	 * Check whether the conditional object is currently needed.
+	 *
+	 * @return bool Whether the conditional object is needed.
+	 */
+	public static function is_needed() {
+
+		return is_admin();
+	}
+
+	/**
+	 * Fetch AMP plugin data.
+	 *
+	 * @return void
+	 */
+	public static function set_plugins() {
+
+		global $wp_filesystem;
+
+		AMPThemes::init_file_system();
+
+		$plugin_json = AMP__DIR__ . '/data/plugins.json';
+
+		if ( ! file_exists( $plugin_json ) ) {
+			return;
+		}
+
+		$json_data       = $wp_filesystem->get_contents( $plugin_json );
+		self::$plugins   = json_decode( $json_data, true );
+		$json_last_error = json_last_error();
+
+		if ( JSON_ERROR_NONE !== $json_last_error ) {
+			self::$plugins = [];
+		}
+	}
+
+	/**
+	 * Adds hooks.
+	 *
+	 * @return void
+	 */
+	public function register() {
+
+		$this->set_plugins();
+		$screen = get_current_screen();
+
+		if ( $screen instanceof \WP_Screen && in_array( $screen->id, [ 'plugins', 'plugin-install' ], true ) ) {
+			add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_scripts' ] );
+		}
+
+		add_filter( 'install_plugins_tabs', [ $this, 'add_tab' ] );
+		add_filter( 'install_plugins_table_api_args_px_enhancing', [ $this, 'tab_args' ] );
+		add_filter( 'plugins_api', [ $this, 'plugins_api' ], 10, 3 );
+		add_filter( 'plugin_install_action_links', [ $this, 'action_links' ], 10, 2 );
+		add_filter( 'plugin_row_meta', [ $this, 'plugin_row_meta' ], 10, 3 );
+
+		add_action( 'install_plugins_px_enhancing', 'display_plugins_table' );
+	}
+
+	/**
+	 * Enqueue style for plugin install page.
+	 *
+	 * @return void
+	 */
+	public function enqueue_scripts() {
+
+		$asset_file   = AMP__DIR__ . '/assets/js/' . self::ASSET_HANDLE . '.asset.php';
+		$asset        = require $asset_file;
+		$dependencies = $asset['dependencies'];
+		$version      = $asset['version'];
+
+		wp_enqueue_script(
+			self::ASSET_HANDLE,
+			amp_get_asset_url( 'js/' . self::ASSET_HANDLE . '.js' ),
+			$dependencies,
+			$version,
+			true
+		);
+
+		wp_enqueue_style(
+			'amp-admin',
+			amp_get_asset_url( 'css/amp-admin.css' ),
+			[],
+			AMP__VERSION
+		);
+
+		$none_wporg = [];
+
+		foreach ( self::$plugins as $plugin ) {
```

Comment:
Per above, this can rather do:

```suggestion
		foreach ( $this->get_plugins() as $plugin ) {
```